proposals/collection-literals.md

```diff
@@ -112,23 +112,46 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
+* To a single dimensional *array type* `T[]`, or a *span type* `System.Span<T>` or `System.ReadOnlySpan<T>`, where for each element `Ei`:
+  * If `Ei` is an *expression element* there is an implicit conversion from `Ei` to `T`.
+  * If `Ei` is a *dictionary element* `Ki:Vi`, then `T` is a type `System.Collections.Generic.KeyValuePair<K, V>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *spread element* with *iteration type* `Si` there is an implicit conversion from `Si` to `T`.
 
-Actual translation of the literal to the corresponding type is defined [below](#collection-literal-translation).
+* To a *type* with an associated *[builder](#construct-methods)* where there is an implicit collection literal conversion from the collection literal to the *span type* of the builder argument.
 
-* Single dimensional arrays (e.g. `T[]`)
-* [*Span types*](#span-types)
-* Types with a suitable [`Construct` method](#construct-methods)
-* Types that implement `System.Collections.IDictionary` - *dictionary [collection initializers](#collection-initializers)*
-* Interface types _`I<TKey, TValue>`_ implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
-* Types that implement `System.Collections.IEnumerable` - [*collection initializers*](#collection-initializers)
-* Interface types _`I<T>`_ implemented by `System.Collections.Generic.List<T>` (e.g. `IEnumerable<T>`, `IList<T>`, `IReadOnlyList<T>`)
+* To a *type* that implements `System.Collections.IDictionary` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter where the parameter has type `System.Int32` and name `capacity`.
+  * For each element `Ei` there is an applicable instance indexer for the key value pair from:
+    * If `Ei` is an *expression element* with type `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from `Ei`.
+    * If `Ei` is a *dictionary element* `Ki:Vi`, the values `Ki` and `Vi`.
+    * If `Ei` is a *spread element* with *iteration type* `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from the *iteration variable*.
+
+* To an *interface type* *`I<K, V>`* where `I<TKey, TValue>` is implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` and where for each element `Ei`:
+  * If `Ei` is an *expression element* then the type of `Ei` is `dynamic`, or the type of `Ei` is a type `System.Collections.Generic.KeyValuePair<Ki, Vi>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *dictionary element* `Ki:Vi` there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
```

Comment:
This phrasing isn't quite right either. You want to say something more like "If ei is a dictionary element ki:vi and there exists implicit conversions from ki to k and vi to v". As written right now, you're bringing those conversions into existence, not querying if they exist.

Reply:
This clause is part of "and where for each element `Ei`:" above. We could use different phrasing, but with "If `Ei` is a dictionary element `Ki:Vi` and there exists implicit conversions from `Ki` to K and `Vi` to `V` ..." it seems like we need to complete the sentence.

Reply:
Updated based on our offline discussion.